http/src/main/resources/init-resources/gce-init.sh

```diff
@@ -140,77 +207,28 @@ INIT_BUCKET_NAME=$(initBucketName)
 CERT_DIRECTORY='/var/certs'
 DOCKER_COMPOSE_FILES_DIRECTORY='/var/docker-compose-files'
 WORK_DIRECTORY='/mnt/disks/work'
+# Toolbox is specific to COS images and is needed to access functionalities like gcloud
+# See https://cloud.google.com/container-optimized-os/docs/how-to/toolbox
 GSUTIL_CMD='docker run --rm -v /var:/var us.gcr.io/cos-cloud/toolbox:v20230714 gsutil'
 GCLOUD_CMD='docker run --rm -v /var:/var us.gcr.io/cos-cloud/toolbox:v20230714 gcloud'
 
+# Welder configuration, Rstudio files are saved every X seconds in the background but Jupyter notebooks are not
 if [ ! -z "$RSTUDIO_DOCKER_IMAGE" ] ; then
   export SHOULD_BACKGROUND_SYNC="true"
 else
   export SHOULD_BACKGROUND_SYNC="false"
 fi
 
+
+# Use specific docker compose command if the container is coming from GCR -> WHY?
```

Comment:
This might be a clue as to why, from the image documentation: https://hub.docker.com/r/cryptopants/docker-compose-gcr

```
This container helps resolve an issue with Container Optimized OS. The OS does not come with docker-compose, and instead expects users to use containerized docker-compose. However the official containerized docker-compose does not include docker-credential-gcr, which is needed to pull Docker images from a private Google Container Registry.

This container uses the official docker-compose as a base, and adds docker-credential-gcr. It can be used just like the normal containerized docker-compose, but can now access gcr.io.
```

So sounds like it just extends `docker/compose` to support GCR. I am not sure what will work with GAR…